app.rb

```diff
@@ -31,19 +32,74 @@
 config_overrides[:port] = ENV['APRUVE_PORT'] unless ENV['APRUVE_PORT'].nil?
 Apruve.configure(ENV['APRUVE_API_KEY'], apruve_environment, config_overrides)
 merchant_id = ENV['APRUVE_MERCHANT_ID']
+corporate_account_id = ENV['APRUVE_CORPORATE_ACCOUNT_ID']
 
 before '/webhook_notify' do
   request.body.rewind
   @webhook_body = request.body.read
 end
 
+get '/offline_order' do
+  erb :offline_orders
+end
+
+get '/customers' do
+  content_type :json
+  send_file File.expand_path('customers.json', settings.public_folder)
+end
+
+get '/shopper_id/:email' do
+  content_type :json
+  begin
+    corporate_account = Apruve::CorporateAccount.find(merchant_id, params['email'])
+    return { shoper_id: corporate_account.customer_uuid } .to_json
+  rescue Apruve::NotFound
+    status 404
+    return ''
+  end
+end
+
+post '/orders' do
+  request.body.rewind
+  payload = JSON.parse request.body.read
+  total_amount_cents = payload['orders'].map { |order|
+    order['subtotal'].to_f
+  }.reduce(:+)
+  total_amount_cents = ((total_amount_cents + payload['shipping'].to_f) * 100).round
+
+  @order = Apruve::Order.new(
+    merchant_id: merchant_id,
+    shopper_id: payload['shopper_id'],
+    currency: 'USD',
+    amount_cents: total_amount_cents,
+    shipping_cents: (payload['shipping'].to_f * 100).round,
+    payment_term: { corporate_account_id: corporate_account_id }
+  )
+
+  payload['orders'].each { |order|
+    @order.order_items << Apruve::OrderItem.new(
+      title: order['name'],
+      description: order['name'],
+      sku: order['sku'],
+      price_ea_cents: (order['price'].to_f * 100).round,
+      quantity: order['quantity'].to_i,
+      price_total_cents: (order['subtotal'].to_f * 100).round
+    )
+  }
+
+  @order.save!
+end
+
 get '/' do
   # Create a payment request and some line items
   @order = Apruve::Order.new(
       merchant_id: merchant_id,
+      shopper_id: 'd03bf60f2e5b8d983e7131a9a1df4cd1',
       currency: 'USD',
       amount_cents: 6000,
-      shipping_cents: 500
+      shipping_cents: 500,
+      payment_term: {corporate_account_id: '8d8c66b9051e5c1b8e7a523e21ee7575'}
+
```

Comment:
It looks like @donnen added a validation to `Apruve::Order` that requires that payment_term be set.  The core classes of the library run validation before the objects get turned into JSON.    That payment term validation doesn't work as a client-side validation, since payment_term is normally set during the checkout process.  I would say that needs to go.